ethereum/jsonrpc/src/main/java/tech/pegasys/pantheon/ethereum/jsonrpc/internal/methods/permissioning/PermAddAccountsToWhitelist.java

```diff
@@ -64,6 +64,8 @@ public JsonRpcResponse response(final JsonRpcRequest request) {
         case ERROR_DUPLICATED_ENTRY:
           return new JsonRpcErrorResponse(
               request.getId(), JsonRpcError.ACCOUNT_WHITELIST_DUPLICATED_ENTRY);
+        case ERROR_WHITELIST_PERSIST_FAIL:
+          return new JsonRpcErrorResponse(request.getId(), JsonRpcError.WHITELIST_PERSIST_FAILURE);
```

Comment:
I feel like this type of error is too fine grained. Assuming we verify at startup that the Pantheon process has to write access to the permissions file, we should assume that any writing errors will be unrecoverable.

That means that we can let it fail as an internal error or something instead of adding one more error message. Thoughts?

Reply:
Makes sense, it's hard to think of a likely case where a node will be able to recover from a state where it can't write to the file.

Reply:
I've left this in because:
- I've added read + write checks on the file when we validate the CLI config
- We've added another condition for when the file is out of sync, but decided to allow the client to continue in this case
- This case is different to the previous case ^